### ISSUE TITLE
Cannot run ad-hoc command when hostname includes utf-8 characters: "UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 5: ordinal not in range(128)"

### ISSUE THREAD

**Issue by jladdjr** [ansible/ansible#15479]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 1.9.6 (stable-1.9 b31e29f124) last updated 2016/04/19 11:11:28 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 6972d0a291) last updated 2016/04/19 11:17:47 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 1220e47f07) last updated 2016/04/19 11:17:52 (GMT -400)
  configured module search path = None
```
##### CONFIGURATION

_ansible.cfg_

```
[ssh_connection]
control_path = %(directory)s/%%h-%%r
```

environment includes:
`ANSIBLE_HOST_KEY_CHECKING=False`
##### OS / ENVIRONMENT

Verified this on:
- Mac OS X 10.11.4
##### SUMMARY

Running the ping command against a host called `Host-䙴栚퉈猟濍㦳짟颒` fails due to a `UnicodeDecodeError`. If I replace this name with `testhost`, the ping command is successful.
##### STEPS TO REPRODUCE

Create the following inventory file:

```
Host-䙴栚퉈猟濍㦳짟颒 ansible_ssh_host=127.0.0.1 ansible_connection=local
testhost ansible_ssh_host=127.0.0.1 ansible_connection=local
```

Try running the ping command against `Host-䙴栚퉈猟濍㦳짟颒` using:
 `ansible Host-䙴栚퉈猟濍㦳짟颒 -i /path/to/inventory -m ping`.

Try running the ping command against `testhost` using:
`ansible testhost -i /path/to/inventory -m ping`
##### EXPECTED RESULTS

In both cases, expect a successful ping:

```
<hostname> | success >> {
    "changed": false,
    "ping": "pong"
}
```
##### ACTUAL RESULTS

Running the command against `Host-䙴栚퉈猟濍㦳짟颒` fails:

```
Host-䙴栚퉈猟濍㦳짟颒 | FAILED => Traceback (most recent call last):
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/runner/__init__.py", line 589, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/runner/__init__.py", line 793, in _executor_internal
    return self._executor_internal_inner(host, self.module_name, self.module_args, inject, port, complex_args=complex_args)
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/runner/__init__.py", line 1125, in _executor_internal_inner
    self.callbacks.on_ok(host, data)
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/callbacks.py", line 389, in on_ok
    self._on_any(host,res)
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/callbacks.py", line 432, in _on_any
    (msg, color) = host_report_msg(host, self.options.module_name, result2, self.options.one_line)
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/callbacks.py", line 330, in host_report_msg
    msg = (regular_generic_msg(hostname, result, oneline, 'success'), 'green')
  File "/Users/jladd/ansible/stable-1.9/lib/ansible/callbacks.py", line 255, in regular_generic_msg
    return "%s | %s >> %s\n" % (hostname, caption, utils.jsonify(result,format=True))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 5: ordinal not in range(128)
```

Running the command against `testhost`, however, is successful.

**Comment by jladdjr** [comment]
Tried running this against other versions of ansible as well (using same steps to reproduce given above):
- On ansible 1.9.6 - get `UnicodeDecodeError` mentioned above.
- On ansible 2.0.1.0 - get `ERROR! Specified hosts options do not match any hosts`
- On ansible 2.0.2.0 - ping is successful:

```
Host-䙴栚퉈猟濍㦳짟颒 | SUCCESS => {
    "changed": false,
    "ping": "pong"
}
```

Based on those results, it seems like there is a fix for this on `2.x`, but not on `1.9.x`.

**Comment by jladdjr** [comment]
Some additional context:

This issue originally surface during Tower testing (see https://github.com/ansible/ansible-tower/issues/1302). Received help from @abadger, who wrote a fix here - https://github.com/ansible/ansible/commit/13b295f1eeb2789d83f158a9901aef8fb62ffcff.

Currently seeing this issue on Tower 2.4.5 + Ansible 2.0.1.0. It seems like the new version of ansible 2.x will resolve this problem. Wanted to make sure customers with 1.9.x also received the fix.

**Comment by jladdjr** [comment]
Touched base with @jimi-c - he mentioned that only critical fixes were being back ported to 1.9. Closing this out.
